File: jobs/reddit_headline.rb
recurring_job('1m') do
  listing = JSON.parse(Net::HTTP.get(URI('https://www.reddit.com/r/todayilearned.json?limit=5')))

  title = "not found"
  score = "0"
  listing['data']['children'].each do |item|
    next if item['data']['stickied']
    title = item['data']['title']
    score = item['data']['score']
    break
  end
  {
    reddit_headline: {text: title},
    reddit_score: {number: score}
  }
end

File: jobs/reddit_headline.js
var request = require("request")
const url = "https://www.reddit.com/r/todayilearned.json?limit=5";

exports.interval = 5000;
exports.promise = function(fulfill, reject) {
  request(url, function (error, response, body) {
    if (!error && response.statusCode == 200) {
      var json = JSON.parse(body);
      for (var i = 0; i < json["data"]["children"].length; i++) {
        var child = json["data"]["children"][i];
        if(child["data"]["stickied"]) {
          continue;
        }
        fulfill({
          reddit_headline: {text: child["data"]["title"]},
          reddit_score: {number: child["data"]["score"]}
        });
        break;
      }
    } else {
      reject(error);
    }
  });
};
